lib/smart_answer/calculators/marriage_abroad_calculator.rb

```diff
@@ -325,10 +325,14 @@ def services_data_for_country_and_partner_sex_and_residency?
     end
 
     def services_data_for_country_and_partner_sex?
-      @services_data.has_key?(ceremony_country) &&
+      services_data_for_country? &&
         @services_data[ceremony_country].has_key?(@sex_of_your_partner)
     end
 
+    def services_data_for_country?
+      @services_data.has_key?(ceremony_country)
+    end
+
```

Comment:
Minor: Why did you decide to extract `@services_data.has_key?(ceremony_country)` and not `@services_data[ceremony_country]`? If the latter defaulted to returning an empty Hash, I think you could get rid of the `if` statement in `services_payment_partial_name` and first condition in `services_data_for_country_and_partner_sex`. I think getting rid of conditionals is generally a good thing.